src/Unosquare.PassCore.PasswordProvider/PasswordChangeProvider.cs

```diff
@@ -126,7 +126,7 @@ public ApiErrorItem PerformPasswordChange(string username, string currentPasswor
             if (principalContext.ValidateCredentials(upn, currentPassword))
                 return true;
 
-            var tmpAuthority = upn?.Split('@').Last();
+            var tmpAuthority = upn?.Contains('@') == true ? upn.Split('@').Last() : string.Empty;

```

Comment:
Just a note regarding this change.
As written in #343, when the user's UPN suffix is different by AD domain:

- UPN suffix = **domain.com**
- AD domain = **domain.local**

...using...

`var tmpAuthority = upn?.Split('@').Last();`

OR 

`var tmpAuthority = upn?.Contains('@') == true ? upn.Split('@').Last() : string.Empty;`

produces the same results: **tmpAuthority = domain.com**; but then, the _LogonUser(...)_ function will return **1326 (ERROR_LOGON_FAILURE)** even if the end-user has "_Must change password at next logon_" flag selected - so it should return **1907 (ERROR_PASSWORD_MUST_CHANGE)** or its password is expired - so it should return **1330 (ERROR_PASSWORD_EXPIRED)**.

According to this **Microsoft Docs** article _[LogonUserA function](https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-logonusera)_:

```
BOOL LogonUserA(
  LPCSTR  lpszUsername,
  LPCSTR  lpszDomain,
  LPCSTR  lpszPassword,
  DWORD   dwLogonType,
  DWORD   dwLogonProvider,
  PHANDLE phToken
);
```

> lpszUsername
> 
> A pointer to a null-terminated string that specifies the name of the user. This is the name of the user account to log on to. If you use the user principal name (UPN) format, User@DNSDomainName, the lpszDomain parameter must be NULL.

Reply:
You are right, I missed that. I will work on this and let you know when it's finished.

Reply:
Thank you very much @Serk352.